File: sdks/php/src/Dialect/PostgresDialect.php
<?php
namespace Autonoma\Sdk\Dialect;

class PostgresDialect implements DialectInterface
{
    public function getName(): string { return 'postgres'; }
    public function supportsReturning(): bool { return true; }
    public function param(int $index): string { return "\${$index}"; }
    public function quoteId(string $name): string { return '"' . $name . '"'; }
    public function tablesSql(string $schema): string { return SqlQueries::replaceSchema(SqlQueries::POSTGRES_TABLES, $schema); }
    public function columnsSql(string $schema): string { return SqlQueries::replaceSchema(SqlQueries::POSTGRES_COLUMNS, $schema); }
    public function primaryKeysSql(string $schema): string { return SqlQueries::replaceSchema(SqlQueries::POSTGRES_PRIMARY_KEYS, $schema); }
    public function foreignKeysSql(string $schema): string { return SqlQueries::replaceSchema(SqlQueries::POSTGRES_FOREIGN_KEYS, $schema); }
    public function enumsSql(string $schema): string { return SqlQueries::POSTGRES_ENUMS; }
}

File: sdks/java/autonoma-sdk/src/main/java/ai/autonoma/sdk/PostgresDialect.java
package ai.autonoma.sdk;

public class PostgresDialect implements Dialect {

    @Override public String name() { return "postgres"; }
    @Override public String param(int index) { return "$" + index; }
    @Override public String quoteId(String name) { return "\"" + name + "\""; }
    @Override public boolean supportsReturning() { return true; }

    @Override
    public String tablesSQL(String schema) {
        return SqlQueries.POSTGRES_TABLES.replace("{{schema}}", schema);
    }

    @Override
    public String columnsSQL(String schema) {
        return SqlQueries.POSTGRES_COLUMNS.replace("{{schema}}", schema);
    }

    @Override
    public String primaryKeysSQL(String schema) {
        return SqlQueries.POSTGRES_PRIMARY_KEYS.replace("{{schema}}", schema);
    }

    @Override
    public String foreignKeysSQL(String schema) {
        return SqlQueries.POSTGRES_FOREIGN_KEYS.replace("{{schema}}", schema);
    }

    @Override
    public String enumsSQL(String schema) {
        return SqlQueries.POSTGRES_ENUMS;
    }
}
